<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>cytobox examples • cytobox</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="cytobox examples">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">cytobox</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.6.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/cytobox.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/fungenomics/cytobox">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>cytobox examples</h1>
            
            <h4 class="date">2019-07-29</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/fungenomics/cytobox/blob/master/vignettes/cytobox.Rmd"><code>vignettes/cytobox.Rmd</code></a></small>
      <div class="hidden name"><code>cytobox.Rmd</code></div>

    </div>

    
    
<p>Load the package</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/library">library</a></span>(cytobox)</code></pre></div>
<p>Here is a quick overview of some of the most useful functions available in <code>cytobox</code> for exploring single cell RNA-seq data stored in a Seurat package. It’s not an exhaustive list of the functions, or all the options, so see the documentation for more details.</p>
<div id="basic-visualizations" class="section level1">
<h1 class="hasAnchor">
<a href="#basic-visualizations" class="anchor"></a>Basic visualizations</h1>
<div id="t-snepca" class="section level2">
<h2 class="hasAnchor">
<a href="#t-snepca" class="anchor"></a>t-SNE/PCA</h2>
<p>Basic t-SNE of a dataset (like <code><a href="https://www.rdocumentation.org/packages/Seurat/topics/TSNEPlot">Seurat::TSNEPlot()</a></code>), or PCA:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="kw"><a href="../reference/plot_dr.html">tsne</a></span>(pbmc)</code></pre></div>
<p><img src="cytobox_files/figure-html/tsne-1.png" width="576"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/plot_dr.html">tsne</a></span>(pbmc, <span class="dt">label_repel =</span> <span class="ot">FALSE</span>)</code></pre></div>
<p><img src="cytobox_files/figure-html/tsne-2.png" width="576"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="kw"><a href="../reference/plot_dr.html">pca</a></span>(pbmc, <span class="dt">label_repel =</span> <span class="ot">FALSE</span>, <span class="dt">point_size =</span> <span class="dv">2</span>)</code></pre></div>
<p><img src="cytobox_files/figure-html/tsne-3.png" width="576"></p>
<p>Colour the cells by a column in <code>seurat@meta.data</code> containing a continuous variable, e.g. the number of UMIs:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/plot_dr.html">tsne</a></span>(pbmc, <span class="dt">colour_by =</span> <span class="st">"nUMI"</span>, <span class="dt">colour_by_type =</span> <span class="st">"continuous"</span>, <span class="dt">colours =</span> viridis<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/viridis/topics/reexports">magma</a></span>(<span class="dv">100</span>))</code></pre></div>
<p><img src="cytobox_files/figure-html/tsne_nUMI-1.png" width="576"></p>
<p>Only plot specified cells, say the cells in clusters 1 or 2, coloured by nUMI:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">cells_keep &lt;-<span class="st"> </span><span class="kw"><a href="../reference/whichCells.html">whichCells</a></span>(pbmc, <span class="dt">clusters =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">1</span>, <span class="dv">2</span>))
cells_keep
<span class="co">#&gt;  [1] "TACGCCACTCCGAA" "CTAAACCTGTGCAT" "CATCATACGGAGCA" "TTACCATGAATCGC"</span>
<span class="co">#&gt;  [5] "ATAGGAGAAACAGA" "GCGCACGACTTTAC" "ACTCGCACGAAAGT" "ATTACCTGCCTTAT"</span>
<span class="co">#&gt;  [9] "CCCAACTGCAATCG" "AAATTCGAATCACG" "CCATCCGATTCGCC" "TCCACTCTGAGCTT"</span>
<span class="co">#&gt; [13] "CTAAACCTCTGACA" "GATAGAGAAGGGTG" "CTAACGGAACCGAT" "AGATATACCCGTAA"</span>
<span class="co">#&gt; [17] "TACTCTGAATCGAC" "GCGCATCTTGCTCC" "GTTGACGATATCGG" "ACAGGTACTGGTGT"</span>
<span class="co">#&gt; [21] "GGCATATGCTTATC" "CATTACACCAACTG" "TAGGGACTGAACTC" "AAGCAAGAGCTTAG"</span>
<span class="co">#&gt; [25] "GGTGGAGATTACTC" "CGTAGCCTGTATGC" "CCTATAACGAGACG" "ATAAGTTGGTACGT"</span>
<span class="co">#&gt; [29] "ACCAGTGAATACCG" "ATTGCACTTGCTTT" "CTAGGTGATGGTTG" "CATGCGCTAGTCAC"</span>
<span class="co">#&gt; [33] "TTGAGGACTACGCA" "CATATAGACTAAGC" "TTTAGCTGTACTCT" "GACATTCTCCACCT"</span>
<span class="co">#&gt; [37] "ACGTGATGCCATGA" "ATTGTAGATTCCCG" "GATAGAGATCACGA" "AATGCGTGGACGGA"</span>
<span class="co">#&gt; [41] "GCGTAAACACGGTT"</span>

<span class="kw"><a href="../reference/plot_dr.html">tsne</a></span>(pbmc,
     <span class="dt">colour_by =</span> <span class="st">"nUMI"</span>,
     <span class="dt">colour_by_type =</span> <span class="st">"continuous"</span>,
     <span class="dt">colours =</span> viridis<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/viridis/topics/reexports">magma</a></span>(<span class="dv">100</span>),
     <span class="co"># Provide a vector of cell names</span>
     <span class="dt">cells =</span> cells_keep,
     <span class="co"># Only label these two clusters</span>
     <span class="dt">clusters_to_label =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">1</span>, <span class="dv">2</span>))</code></pre></div>
<p><img src="cytobox_files/figure-html/tsne_nUMI_select-1.png" width="576"></p>
<p>Colour the cells by a column in <code>seurat@meta.data</code> containing a discrete variable, e.g. here, the resolution 1.0 cluster assignments:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/plot_dr.html">tsne</a></span>(pbmc, <span class="dt">colour_by =</span> <span class="st">"res.1"</span>, <span class="dt">colour_by_type =</span> <span class="st">"discrete"</span>,
     <span class="co"># Getting some colours from a Color Brewer palette</span>
     <span class="dt">colours =</span> RColorBrewer<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/RColorBrewer/topics/ColorBrewer">brewer.pal</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/length">length</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/unique">unique</a></span>(pbmc<span class="op">@</span>meta.data<span class="op">$</span>res.<span class="dv">1</span>)), <span class="st">"Dark2"</span>))</code></pre></div>
<p><img src="cytobox_files/figure-html/tsne_res.1-1.png" width="576"></p>
</div>
<div id="highlight-cluster-in-t-sne-space" class="section level2">
<h2 class="hasAnchor">
<a href="#highlight-cluster-in-t-sne-space" class="anchor"></a>Highlight cluster in t-SNE space</h2>
<p>The last example demonstrates how to only plot specific cells, using the string “none” - this is more effective than making the default colour “white”, because it avoids cells coloured white being plot on top of cells we have coloured, which becomes important in areas of high density of cells.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="co"># Highlight cluster 3 on the tSNE plot</span>
<span class="kw"><a href="../reference/highlight.html">highlight</a></span>(pbmc, <span class="dv">3</span>)</code></pre></div>
<p><img src="cytobox_files/figure-html/highlight-1.png" width="576"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="co"># Pass additional arguments to cytobox::tsne</span>
<span class="kw"><a href="../reference/highlight.html">highlight</a></span>(pbmc, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">2</span>, <span class="dv">3</span>), <span class="dt">label =</span> <span class="ot">FALSE</span>, <span class="dt">title =</span> <span class="st">"Test highlight"</span>)</code></pre></div>
<p><img src="cytobox_files/figure-html/highlight-2.png" width="576"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="co"># Change default colour</span>
<span class="kw"><a href="../reference/highlight.html">highlight</a></span>(pbmc, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">2</span>, <span class="dv">3</span>), <span class="dt">default_colour =</span> <span class="st">"lightblue"</span>)</code></pre></div>
<p><img src="cytobox_files/figure-html/highlight-3.png" width="576"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="co"># Specify the colours to highlight the clusters with</span>
<span class="kw"><a href="../reference/highlight.html">highlight</a></span>(pbmc, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">2</span>, <span class="dv">3</span>), <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"red"</span>, <span class="st">"blue"</span>))</code></pre></div>
<p><img src="cytobox_files/figure-html/highlight-4.png" width="576"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="co"># Don't plot cells in other clusters</span>
<span class="kw"><a href="../reference/highlight.html">highlight</a></span>(pbmc, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">2</span>, <span class="dv">3</span>), <span class="dt">default_colour =</span> <span class="st">"none"</span>)</code></pre></div>
<p><img src="cytobox_files/figure-html/highlight-5.png" width="576"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="co"># PCA space</span>
<span class="kw"><a href="../reference/highlight.html">highlight</a></span>(pbmc, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">2</span>, <span class="dv">3</span>), <span class="dt">default_colour =</span> <span class="st">"none"</span>, <span class="dt">reduction =</span> <span class="st">"pca"</span>)</code></pre></div>
<p><img src="cytobox_files/figure-html/highlight-6.png" width="576"></p>
</div>
</div>
<div id="computed-visualizations" class="section level1">
<h1 class="hasAnchor">
<a href="#computed-visualizations" class="anchor"></a>Computed visualizations</h1>
<p>Plot a t-SNE coloured by mean marker expression, feature-plot style:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="kw"><a href="../reference/tsneByMeanMarkerExpression.html">tsneByMeanMarkerExpression</a></span>(pbmc, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"IL32"</span>, <span class="st">"CD2"</span>))</code></pre></div>
<p><img src="cytobox_files/figure-html/tsneByMeanMarkerExpression-1.png" width="576"></p>
<p>Plot a t-SNE coloured by percentiles marker expression</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="kw"><a href="../reference/tsneByPercentileMarkerExpression.html">tsneByPercentileMarkerExpression</a></span>(pbmc, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"IL32"</span>, <span class="st">"CD2"</span>))</code></pre></div>
<p><img src="cytobox_files/figure-html/tsneByPercentilesMarkerExpression-1.png" width="576"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="co"># No labels</span>
<span class="kw"><a href="../reference/tsneByPercentileMarkerExpression.html">tsneByPercentileMarkerExpression</a></span>(pbmc, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"IL32"</span>, <span class="st">"CD2"</span>), <span class="dt">label =</span> <span class="ot">FALSE</span>)</code></pre></div>
<p><img src="cytobox_files/figure-html/tsneByPercentilesMarkerExpression-2.png" width="576"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="co"># Use values in legend instead</span>
<span class="kw"><a href="../reference/tsneByPercentileMarkerExpression.html">tsneByPercentileMarkerExpression</a></span>(pbmc, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"IL32"</span>, <span class="st">"CD2"</span>), <span class="dt">label =</span> <span class="ot">FALSE</span>,
                                 <span class="dt">legend_options =</span> <span class="st">"values"</span>)</code></pre></div>
<p><img src="cytobox_files/figure-html/tsneByPercentilesMarkerExpression-3.png" width="576"></p>
<p>Dashboard plot, which gives a bit more distributional information than tSNE:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="kw"><a href="../reference/tsneByPercentileMarkerExpression.html">tsneByPercentileMarkerExpression</a></span>(pbmc, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"IL32"</span>, <span class="st">"CD2"</span>), <span class="dt">label =</span> <span class="ot">TRUE</span>, <span class="dt">extra =</span> <span class="ot">TRUE</span>, <span class="dt">verbose =</span> <span class="ot">TRUE</span>)
<span class="co">#&gt; Computing percentiles...</span>
<span class="co">#&gt; Constructing percentiles plot...</span>
<span class="co">#&gt; Computing frequencies...</span>
<span class="co">#&gt; Constructing bar plot...</span>
<span class="co">#&gt; Computing means...</span>
<span class="co">#&gt; Constructing ridge plot...</span>
<span class="co">#&gt; Combining plots...</span>
<span class="co">#&gt; Picking joint bandwidth of 0.749</span></code></pre></div>
<p><img src="cytobox_files/figure-html/dashboard-1.png" width="768"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="co"># This function is a shortcut for the above</span>
<span class="kw"><a href="../reference/tsneByPercentileMarkerExpression.html">dashboard</a></span>(pbmc, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"IL32"</span>, <span class="st">"CD2"</span>), <span class="dt">title =</span> <span class="st">"test genes"</span>)
<span class="co">#&gt; Picking joint bandwidth of 0.749</span></code></pre></div>
<p><img src="cytobox_files/figure-html/dashboard-2.png" width="768"></p>
<p>Notes:</p>
<ul>
<li>This doesn’t look good with few clusters; for a real sample with ~ 15 clusters, a good set of dimensions is width = 14, height = 8</li>
<li>Known issue: the dashboard plots throw a big warning related to the unicode symbols. Do warning = FALSE in the chunk options to not print it.</li>
</ul>
<p>Feature plot, akin to <code><a href="https://www.rdocumentation.org/packages/Seurat/topics/FeaturePlot">Seurat::FeaturePlot()</a></code></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="kw"><a href="../reference/tsneByPercentileMarkerExpression.html">tsneByPercentileMarkerExpression</a></span>(pbmc, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"IL32"</span>, <span class="st">"CD2"</span>), <span class="dt">label_repel =</span> <span class="ot">FALSE</span>, <span class="dt">palette =</span> <span class="st">"redgrey"</span>, <span class="dt">alpha =</span> <span class="ot">FALSE</span>, <span class="dt">point_size =</span> <span class="dv">1</span>, <span class="dt">legend =</span> <span class="ot">FALSE</span>)</code></pre></div>
<p><img src="cytobox_files/figure-html/feature-1.png" width="384"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="co"># This function is a shortcut for the above, except by default it makes one plot</span>
<span class="co"># per gene (set per_gene = FALSE like below to have one plot with the summary statistic)</span>
<span class="kw"><a href="../reference/tsneByMeanMarkerExpression.html">feature</a></span>(pbmc, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"IL32"</span>, <span class="st">"CD2"</span>), <span class="dt">per_gene =</span> <span class="ot">FALSE</span>)</code></pre></div>
<p><img src="cytobox_files/figure-html/feature-2.png" width="384"></p>
<p>Feature plots for individual genes:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="kw"><a href="../reference/tsneByMeanMarkerExpression.html">feature</a></span>(pbmc, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"IL32"</span>, <span class="st">"CD2"</span>, <span class="st">"MS4A1"</span>), <span class="dt">per_gene =</span> <span class="ot">TRUE</span>)</code></pre></div>
<p><img src="cytobox_files/figure-html/feature_indiv-1.png" width="1152"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/tsneByMeanMarkerExpression.html">feature</a></span>(pbmc, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"IL32"</span>, <span class="st">"CD2"</span>, <span class="st">"MS4A1"</span>), <span class="dt">per_gene =</span> <span class="ot">TRUE</span>, <span class="dt">statistic =</span> <span class="st">"mean"</span>)</code></pre></div>
<p><img src="cytobox_files/figure-html/feature_indiv-2.png" width="1152"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r"><span class="kw"><a href="../reference/tsneByMeanMarkerExpression.html">feature</a></span>(pbmc, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"IL32"</span>, <span class="st">"CD2"</span>, <span class="st">"MS4A1"</span>), <span class="dt">per_gene =</span> <span class="ot">TRUE</span>, <span class="dt">statistic =</span> <span class="st">"mean"</span>, <span class="dt">hide_ticks =</span> <span class="ot">TRUE</span>)</code></pre></div>
<p><img src="cytobox_files/figure-html/feature_indiv-3.png" width="1152"></p>
<div id="violins" class="section level2">
<h2 class="hasAnchor">
<a href="#violins" class="anchor"></a>Violins</h2>
<p>Violin plot for a set of genes (like <code><a href="https://www.rdocumentation.org/packages/Seurat/topics/VlnPlot">Seurat::VlnPlot()</a></code>)</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="kw"><a href="../reference/vln.html">vln</a></span>(pbmc, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"IL32"</span>, <span class="st">"MS4A1"</span>), <span class="dt">facet_by =</span> <span class="st">"gene"</span>)</code></pre></div>
<p><img src="cytobox_files/figure-html/vln-1.png" width="480"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="co"># Inverted, one plot per cluster (the default)</span>
<span class="kw"><a href="../reference/vln.html">vln</a></span>(pbmc, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"IL32"</span>, <span class="st">"MS4A1"</span>), <span class="dt">facet_by =</span> <span class="st">"cluster"</span>)</code></pre></div>
<p><img src="cytobox_files/figure-html/vln-2.png" width="480"></p>
</div>
<div id="violin-grid" class="section level2">
<h2 class="hasAnchor">
<a href="#violin-grid" class="anchor"></a>Violin grid</h2>
<p>Panel of tiny violin plots for a set of genes, as in Figure 5D in the drop-seq paper. See docs for default ordering method.</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="co"># Use the first 15 genes in the data</span>
<span class="kw"><a href="../reference/vlnGrid.html">vlnGrid</a></span>(pbmc, <span class="dt">genes =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/head">head</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">rownames</a></span>(pbmc<span class="op">@</span>data), <span class="dv">15</span>))</code></pre></div>
<p><img src="cytobox_files/figure-html/vlnGrid-1.png" width="672"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="co"># Specify order</span>
<span class="kw"><a href="../reference/vlnGrid.html">vlnGrid</a></span>(pbmc, <span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/head">head</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">rownames</a></span>(pbmc<span class="op">@</span>data), <span class="dv">15</span>), <span class="dt">order =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">0</span>, <span class="dv">3</span>, <span class="dv">2</span>, <span class="dv">1</span>))</code></pre></div>
<p><img src="cytobox_files/figure-html/vlnGrid-2.png" width="672"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="co"># Plot a subset of clusters</span>
<span class="kw"><a href="../reference/vlnGrid.html">vlnGrid</a></span>(pbmc, <span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/head">head</a></span>(<span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/colnames">rownames</a></span>(pbmc<span class="op">@</span>data), <span class="dv">15</span>), <span class="dt">subset_clusters =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">0</span>, <span class="dv">1</span>))</code></pre></div>
<p><img src="cytobox_files/figure-html/vlnGrid-3.png" width="672"></p>
</div>
</div>
<div id="two-sample-plots" class="section level1">
<h1 class="hasAnchor">
<a href="#two-sample-plots" class="anchor"></a>Two-sample plots</h1>
<p>Pariwise violin plot</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="kw"><a href="../reference/pairwiseVln.html">pairwiseVln</a></span>(pbmc, markers_pbmc, pbmc, <span class="dt">sample_names =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"PBMC1"</span>, <span class="st">"PBMC2"</span>))</code></pre></div>
<p><img src="cytobox_files/figure-html/pariwiseVln-1.png" width="576"></p>
<div id="heatmaps-of-marker-overlap" class="section level2">
<h2 class="hasAnchor">
<a href="#heatmaps-of-marker-overlap" class="anchor"></a>Heatmaps of marker overlap</h2>
<p>We may be interested in shared markers between clusters in two samples. When calculating this value, we can use as the denominator of the equation either the number of markers in sample 1 or the number of markers in sample 2. This function provides a few options for what to plot in the heatmap, in the <code>mode</code> argument:</p>
<ul>
<li>“s1”: Plot the percentage of sample 1 markers which are also markers in sample 2</li>
<li>“s2”: Plot the percentage of sample 2 markers which are also markers in sample 1</li>
<li>“min”: Plot the minimum of the above percentages</li>
<li>“max”: Plot the maximum of the above percentages</li>
<li>“both”: Plot two heatmaps, one with the percentage of sample 1 markers which are also markers in sample 2, and one the percentage of sample 2 markers which are also markers in sample 1</li>
</ul>
<p>By default, the function plots the minimum (here, comparing one dataset to itself):</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="kw"><a href="../reference/heatmapPercentMarkerOverlap.html">heatmapPercentMarkerOverlap</a></span>(markers_pbmc, markers_pbmc,
                            <span class="dt">sample_names =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"PBMC1"</span>, <span class="st">"PBMC2"</span>),
                            <span class="co"># The default for this argument is set to the column name</span>
                            <span class="co"># of output of our pipeline, which is 'external_gene_name'</span>
                            <span class="dt">marker_col =</span> <span class="st">"gene"</span>)</code></pre></div>
<p><img src="cytobox_files/figure-html/heatmapPercentMarkerOverlap-1.png" width="480"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="co"># Change the palette: give a set of colours to interpolate, from low to high</span>
<span class="kw"><a href="../reference/heatmapPercentMarkerOverlap.html">heatmapPercentMarkerOverlap</a></span>(markers_pbmc, markers_pbmc,
                            <span class="dt">sample_names =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"PBMC1"</span>, <span class="st">"PBMC2"</span>),
                            <span class="dt">marker_col =</span> <span class="st">"gene"</span>,
                            <span class="dt">palette =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"white"</span>, <span class="st">"red"</span>))</code></pre></div>
<p><img src="cytobox_files/figure-html/heatmapPercentMarkerOverlap-2.png" width="480"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="co"># You can use RColorBrewer palettes like this</span>
<span class="kw"><a href="../reference/heatmapPercentMarkerOverlap.html">heatmapPercentMarkerOverlap</a></span>(markers_pbmc, markers_pbmc,
                            <span class="dt">sample_names =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"PBMC1"</span>, <span class="st">"PBMC2"</span>),
                            <span class="dt">marker_col =</span> <span class="st">"gene"</span>,
                            <span class="dt">palette =</span> RColorBrewer<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/RColorBrewer/topics/ColorBrewer">brewer.pal</a></span>(<span class="dv">8</span>, <span class="st">"OrRd"</span>))</code></pre></div>
<p><img src="cytobox_files/figure-html/heatmapPercentMarkerOverlap-3.png" width="480"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="co"># Different label colour</span>
<span class="kw"><a href="../reference/heatmapPercentMarkerOverlap.html">heatmapPercentMarkerOverlap</a></span>(markers_pbmc, markers_pbmc,
                            <span class="dt">sample_names =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"PBMC1"</span>, <span class="st">"PBMC2"</span>),
                            <span class="dt">marker_col =</span> <span class="st">"gene"</span>,
                            <span class="dt">palette =</span> RColorBrewer<span class="op">::</span><span class="kw"><a href="https://www.rdocumentation.org/packages/RColorBrewer/topics/ColorBrewer">brewer.pal</a></span>(<span class="dv">8</span>, <span class="st">"OrRd"</span>),
                            <span class="dt">label_colour =</span> <span class="st">"black"</span>)</code></pre></div>
<p><img src="cytobox_files/figure-html/heatmapPercentMarkerOverlap-4.png" width="480"></p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="co"># Plot the max instead</span>
<span class="kw"><a href="../reference/heatmapPercentMarkerOverlap.html">heatmapPercentMarkerOverlap</a></span>(markers_pbmc, markers_pbmc, <span class="dt">mode =</span> <span class="st">"max"</span>,
                            <span class="dt">sample_names =</span> <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"PBMC1"</span>, <span class="st">"PBMC2"</span>),
                            <span class="dt">marker_col =</span> <span class="st">"gene"</span>)</code></pre></div>
<p><img src="cytobox_files/figure-html/heatmapPercentMarkerOverlap-5.png" width="480"></p>
<p>Plot both:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="kw"><a href="../reference/heatmapPercentMarkerOverlap.html">heatmapPercentMarkerOverlap</a></span>(markers_pbmc, markers_pbmc, <span class="dt">marker_col =</span> <span class="st">"gene"</span>, <span class="dt">mode =</span> <span class="st">"both"</span>)</code></pre></div>
<p><img src="cytobox_files/figure-html/heatmap_both-1.png" width="960"></p>
</div>
</div>
<div id="utilities" class="section level1">
<h1 class="hasAnchor">
<a href="#utilities" class="anchor"></a>Utilities</h1>
<p>Subset data by gene and cluster:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/head">head</a></span>(<span class="kw"><a href="../reference/fetchData.html">fetchData</a></span>(pbmc, <span class="st">"IL32"</span>, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">1</span>, <span class="dv">2</span>)))
<span class="co">#&gt;   IL32</span>
<span class="co">#&gt; 1    0</span>
<span class="co">#&gt; 2    0</span>
<span class="co">#&gt; 3    0</span>
<span class="co">#&gt; 4    0</span>
<span class="co">#&gt; 5    0</span>
<span class="co">#&gt; 6    0</span>
<span class="kw"><a href="https://www.rdocumentation.org/packages/utils/topics/head">head</a></span>(<span class="kw"><a href="../reference/fetchData.html">fetchData</a></span>(pbmc, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"IL32"</span>, <span class="st">"MS4A1"</span>), <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="dv">1</span>, <span class="dv">2</span>), <span class="dt">return_cluster =</span> <span class="ot">TRUE</span>, <span class="dt">return_cell =</span> <span class="ot">TRUE</span>))
<span class="co">#&gt;             Cell Cluster    MS4A1 IL32</span>
<span class="co">#&gt; 1 TACGCCACTCCGAA       1 5.313305    0</span>
<span class="co">#&gt; 2 CTAAACCTGTGCAT       1 5.190573    0</span>
<span class="co">#&gt; 3 CATCATACGGAGCA       1 5.537982    0</span>
<span class="co">#&gt; 4 TTACCATGAATCGC       1 0.000000    0</span>
<span class="co">#&gt; 5 ATAGGAGAAACAGA       1 0.000000    0</span>
<span class="co">#&gt; 6 GCGCACGACTTTAC       1 0.000000    0</span></code></pre></div>
<p><code>cytobox</code> tells you if you try to plot genes which are undetected in the data:</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
<span class="kw"><a href="../reference/tsneByMeanMarkerExpression.html">tsneByMeanMarkerExpression</a></span>(pbmc, <span class="kw"><a href="https://www.rdocumentation.org/packages/base/topics/c">c</a></span>(<span class="st">"MS4A1"</span>, <span class="st">"foo"</span>))
<span class="co">#&gt; </span><span class="al">NOTE</span><span class="co">: [foo] undetected in pbmc</span></code></pre></div>
<p><img src="cytobox_files/figure-html/undetected-1.png" width="700"></p>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li>
<a href="#basic-visualizations">Basic visualizations</a><ul class="nav nav-pills nav-stacked">
<li><a href="#t-snepca">t-SNE/PCA</a></li>
      <li><a href="#highlight-cluster-in-t-sne-space">Highlight cluster in t-SNE space</a></li>
      </ul>
</li>
      <li>
<a href="#computed-visualizations">Computed visualizations</a><ul class="nav nav-pills nav-stacked">
<li><a href="#violins">Violins</a></li>
      <li><a href="#violin-grid">Violin grid</a></li>
      </ul>
</li>
      <li>
<a href="#two-sample-plots">Two-sample plots</a><ul class="nav nav-pills nav-stacked">
<li><a href="#heatmaps-of-marker-overlap">Heatmaps of marker overlap</a></li>
      </ul>
</li>
      <li><a href="#utilities">Utilities</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by Selin Jessa, Alexis Blanchet-Cohen, Yixing Hu, Claudia L. Kleinman.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
</div>

  

  </body>
</html>
