<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Colour cells in t-SNE or PCA space by gene expression — tsneByMeanMarkerExpression • cytobox</title>

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



<meta property="og:title" content="Colour cells in t-SNE or PCA space by gene expression — tsneByMeanMarkerExpression" />

<meta property="og:description" content="Plot a low-dimensional embedding of the cells,
coloured by expression of a gene, or mean expression of a group of marker
genes. Defaults to t-SNE space, but see the reduction argument for
how to plot in PCA space instead. This function is based on Seurat::FeaturePlot." />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">cytobox</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.6.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/cytobox.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/extra/feature_plots.html">Highly-customizable feature plots</a>
    </li>
  </ul>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/fungenomics/cytobox">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Colour cells in t-SNE or PCA space by gene expression</h1>
    <small class="dont-index">Source: <a href='https://github.com/fungenomics/cytobox/blob/master/R/feature_plots.R'><code>R/feature_plots.R</code></a></small>
    <div class="hidden name"><code>tsneByMeanMarkerExpression.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>Plot a low-dimensional embedding of the cells,
coloured by expression of a gene, or mean expression of a group of marker
genes. Defaults to t-SNE space, but see the <code>reduction</code> argument for
how to plot in PCA space instead. This function is based on <code><a href='https://www.rdocumentation.org/packages/Seurat/topics/FeaturePlot'>Seurat::FeaturePlot</a></code>.</p>
    
    </div>

    <pre class="usage"><span class='fu'>tsneByMeanMarkerExpression</span>(<span class='no'>seurat</span>, <span class='no'>genes</span>, <span class='kw'>reduction</span> <span class='kw'>=</span> <span class='st'>"tsne"</span>,
  <span class='kw'>label</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>palette</span> <span class='kw'>=</span> <span class='st'>"redgrey"</span>, <span class='kw'>title</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>alpha</span> <span class='kw'>=</span> <span class='fl'>0.6</span>,
  <span class='kw'>label_repel</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>label_size</span> <span class='kw'>=</span> <span class='fl'>4</span>, <span class='kw'>legend</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='kw'>hide_ticks</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>hide_axes</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>limits</span> <span class='kw'>=</span> <span class='kw'>NULL</span>,
  <span class='kw'>label_short</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>dim1</span> <span class='kw'>=</span> <span class='fl'>1</span>, <span class='kw'>dim2</span> <span class='kw'>=</span> <span class='fl'>2</span>, <span class='kw'>return_df</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>point_size</span> <span class='kw'>=</span> <span class='fl'>1</span>)

<span class='fu'>feature</span>(<span class='no'>seurat</span>, <span class='no'>genes</span>, <span class='kw'>per_gene</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>statistic</span> <span class='kw'>=</span> <span class='st'>"mean"</span>,
  <span class='kw'>label</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>palette</span> <span class='kw'>=</span> <span class='st'>"redgrey"</span>, <span class='kw'>label_repel</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>label_size</span> <span class='kw'>=</span> <span class='fl'>4</span>, <span class='kw'>label_short</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>legend</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>title</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>reduction</span> <span class='kw'>=</span> <span class='st'>"tsne"</span>, <span class='kw'>limits</span> <span class='kw'>=</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/c'>c</a></span>(<span class='fl'>NA</span>, <span class='fl'>NA</span>), <span class='kw'>dim1</span> <span class='kw'>=</span> <span class='fl'>1</span>,
  <span class='kw'>dim2</span> <span class='kw'>=</span> <span class='fl'>2</span>, <span class='kw'>alpha</span> <span class='kw'>=</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/ifelse'>ifelse</a></span>(<span class='no'>statistic</span> <span class='kw'>==</span> <span class='st'>"percentiles"</span>, <span class='fl'>FALSE</span>, <span class='fl'>0.6</span>),
  <span class='kw'>point_size</span> <span class='kw'>=</span> <span class='fl'>0.5</span>, <span class='kw'>ncol</span> <span class='kw'>=</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/ifelse'>ifelse</a></span>(<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/length'>length</a></span>(<span class='no'>genes</span>) <span class='kw'>==</span> <span class='fl'>1</span>, <span class='fl'>1</span>,
  <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/ifelse'>ifelse</a></span>(<span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/length'>length</a></span>(<span class='no'>genes</span>) <span class='kw'>%in%</span> <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/c'>c</a></span>(<span class='fl'>2</span>, <span class='fl'>4</span>), <span class='fl'>2</span>, <span class='fl'>3</span>)), <span class='kw'>hide_ticks</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='kw'>hide_axes</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>seurat</th>
      <td><p>Seurat object, where dimensionality reduction has been applied,
i.e. (after applying Seurat::RunPCA() or Seurat::RunTSNE() to the object)</p></td>
    </tr>
    <tr>
      <th>genes</th>
      <td><p>String or character vector specifying gene(s) to use</p></td>
    </tr>
    <tr>
      <th>reduction</th>
      <td><p>String specifying the dimensionality reduction to use,
retrieves t-SNE by default. This should match the names of the elements of
the list seurat@dr, so it will typically be one of "pca" or "tsne".
Default: "tsne"</p></td>
    </tr>
    <tr>
      <th>label</th>
      <td><p>Logical, whether to label clusters on the plot. Default: TRUE.</p></td>
    </tr>
    <tr>
      <th>palette</th>
      <td><p>String or character vector. If a string,
one of "viridis", "blues", or "redgrey", specifying which gradient
palette to use. Otherwise, a character vector of colours (from low to high)
to interpolate to create the scael. Default: redgrey.</p></td>
    </tr>
    <tr>
      <th>title</th>
      <td><p>(Optional) String specifying the plot title</p></td>
    </tr>
    <tr>
      <th>alpha</th>
      <td><p>Numeric, fixed alpha for points. Default: 0.6</p></td>
    </tr>
    <tr>
      <th>label_repel</th>
      <td><p>Logical, if <code>label</code> is TRUE, whether to plot cluster
labels repelled from the center, on a slightly transparent white background and
with an arrow pointing to the cluster center. If FALSE, simply plot the
cluster label at the cluster center. Default: TRUE.</p></td>
    </tr>
    <tr>
      <th>label_size</th>
      <td><p>Numeric, controls the size of text labels. Default: 4.</p></td>
    </tr>
    <tr>
      <th>legend</th>
      <td><p>Logical, whether or not to plot legend. Default: TRUE</p></td>
    </tr>
    <tr>
      <th>hide_ticks</th>
      <td><p>Logical, whether to hide axis ticks. Default: FALSE</p></td>
    </tr>
    <tr>
      <th>hide_axes</th>
      <td><p>Logical, whether to hide axis labels. Default: TRUE</p></td>
    </tr>
    <tr>
      <th>limits</th>
      <td><p>(Optional) A numeric vector of length two providing the limits to
use for the colour scale (documentation
from <code>continous_scale</code>. Default: 0 and max of the data.</p></td>
    </tr>
    <tr>
      <th>label_short</th>
      <td><p>(Optional/Experimental!!) Logical, if TRUE, assumes clusters
(at seurat@ident) consist of a prefix and a suffix separated by a non-alpha
numeric character (<code>"[^[:alnum:]]+"</code>), and tries to separate these names
and only plot the prefix, for shorter labels and a cleaner plot. Default: FALSE.</p></td>
    </tr>
    <tr>
      <th>dim1</th>
      <td><p>Numeric, index of dimension from <code>reduction</code> to plot on
the x-axis. e.g. to plot the 3rd PC on the x-axis, pass 3. Default: 1.</p></td>
    </tr>
    <tr>
      <th>dim2</th>
      <td><p>Numeric, like <code>dim2</code>, but for the y-axis. Default: 2.</p></td>
    </tr>
    <tr>
      <th>point_size</th>
      <td><p>Numeric, size of points in scatterplot. Default: 1. (A smaller
value around 0.5 is better for plots which will be viewed at small scale.)</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A ggplot object</p>
    
    <h2 class="hasAnchor" id="functions"><a class="anchor" href="#functions"></a>Functions</h2>

    <ul>
<li><p><code>feature</code>: Shortcut function for plotting mean expression</p></li>
</ul>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'>tsneByMeanMarkerExpression</span>(<span class='no'>pbmc</span>, <span class='st'>"IL32"</span>)</div><div class='img'><img src='tsneByMeanMarkerExpression-1.png' alt='' width='700' height='433' /></div><div class='input'><span class='fu'>tsneByMeanMarkerExpression</span>(<span class='no'>pbmc</span>, <span class='fu'><a href='https://www.rdocumentation.org/packages/base/topics/c'>c</a></span>(<span class='st'>"IL32"</span>, <span class='st'>"CD2"</span>), <span class='kw'>reduction</span> <span class='kw'>=</span> <span class='st'>"pca"</span>)</div><div class='img'><img src='tsneByMeanMarkerExpression-2.png' alt='' width='700' height='433' /></div><div class='input'><span class='fu'>tsneByMeanMarkerExpression</span>(<span class='no'>pbmc</span>, <span class='st'>"IL32"</span>, <span class='kw'>reduction</span> <span class='kw'>=</span> <span class='st'>"pca"</span>, <span class='kw'>dim1</span> <span class='kw'>=</span> <span class='fl'>1</span>, <span class='kw'>dim2</span> <span class='kw'>=</span> <span class='fl'>3</span>)</div><div class='img'><img src='tsneByMeanMarkerExpression-3.png' alt='' width='700' height='433' /></div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#value">Value</a></li>

      <li><a href="#functions">Functions</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

    <h2>Author</h2>
    <p>Selin Jessa</p>
  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Selin Jessa, Alexis Blanchet-Cohen, Yixing Hu, Claudia L. Kleinman.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
   </div>

  

  </body>
</html>

