<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Colour cells in t-SNE space by percentile gene expression — tsneByPercentileMarkerExpression • cytobox</title>

<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous" />
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css" integrity="sha256-eZrrJcwDc/3uDhsdt61sL2oOBY362qM3lon1gyExkL0=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script>

<!-- sticky kit -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>



<meta property="og:title" content="Colour cells in t-SNE space by percentile gene expression — tsneByPercentileMarkerExpression" />

<meta property="og:description" content="Plot a low-dimensional embedding of the cells,
coloured by expression percentile of a gene, of the total expression of a
group of marker genes." />
<meta name="twitter:card" content="summary" />



<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->


  </head>

  <body>
    <div class="container template-reference-topic">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">cytobox</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.6.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../articles/cytobox.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/extra/feature_plots.html">Highly-customizable feature plots</a>
    </li>
  </ul>
</li>
      </ul>
      
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/fungenomics/cytobox">
    <span class="fa fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      
      </header>

<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Colour cells in t-SNE space by percentile gene expression</h1>
    <small class="dont-index">Source: <a href='https://github.com/fungenomics/cytobox/blob/master/R/feature_plots.R'><code>R/feature_plots.R</code></a></small>
    <div class="hidden name"><code>tsneByPercentileMarkerExpression.Rd</code></div>
    </div>

    <div class="ref-description">
    
    <p>Plot a low-dimensional embedding of the cells,
coloured by expression percentile of a gene, of the total expression of a
group of marker genes.</p>
    
    </div>

    <pre class="usage"><span class='fu'>tsneByPercentileMarkerExpression</span>(<span class='no'>seurat</span>, <span class='no'>genes</span>, <span class='kw'>reduction</span> <span class='kw'>=</span> <span class='st'>"tsne"</span>,
  <span class='kw'>label</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>palette</span> <span class='kw'>=</span> <span class='st'>"blues"</span>, <span class='kw'>title</span> <span class='kw'>=</span> <span class='kw'>NULL</span>, <span class='kw'>alpha</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>,
  <span class='kw'>legend</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>legend_options</span> <span class='kw'>=</span> <span class='st'>"percentiles"</span>, <span class='kw'>point_size</span> <span class='kw'>=</span> <span class='fl'>1</span>,
  <span class='kw'>label_repel</span> <span class='kw'>=</span> <span class='fl'>TRUE</span>, <span class='kw'>label_size</span> <span class='kw'>=</span> <span class='fl'>4</span>, <span class='kw'>extra</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>verbose</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>hide_ticks</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>, <span class='kw'>label_short</span> <span class='kw'>=</span> <span class='fl'>FALSE</span>,
  <span class='kw'>dim1</span> <span class='kw'>=</span> <span class='fl'>1</span>, <span class='kw'>dim2</span> <span class='kw'>=</span> <span class='fl'>2</span>)</pre>
    
    <h2 class="hasAnchor" id="arguments"><a class="anchor" href="#arguments"></a>Arguments</h2>
    <table class="ref-arguments">
    <colgroup><col class="name" /><col class="desc" /></colgroup>
    <tr>
      <th>seurat</th>
      <td><p>Seurat object, where dimensionality reduction has been applied,
i.e. (after applying Seurat::RunPCA() or Seurat::RunTSNE() to the object)</p></td>
    </tr>
    <tr>
      <th>genes</th>
      <td><p>String or character vector specifying gene(s) to use</p></td>
    </tr>
    <tr>
      <th>label</th>
      <td><p>Logical, whether to label clusters on the plot. Default: TRUE.</p></td>
    </tr>
    <tr>
      <th>palette</th>
      <td><p>String or character vector. If a string,
one of "viridis", "blues", or "redgrey", specifying which gradient
palette to use. Otherwise, a character vector of colours (from low to high)
to interpolate to create the scael. Default: "blues".</p></td>
    </tr>
    <tr>
      <th>title</th>
      <td><p>(Optional) string used as title for the plot.</p></td>
    </tr>
    <tr>
      <th>alpha</th>
      <td><p>Logical, whether to vary the alpha (point opacity) with percentile
group to highlight cells in the top percentiles.
If FALSE, sets a fixed opacity of 0.8. Default: TRUE.</p></td>
    </tr>
    <tr>
      <th>legend</th>
      <td><p>Logical, whether to plot the legend. Default: FALSE.</p></td>
    </tr>
    <tr>
      <th>legend_options</th>
      <td><p>String, "percentiles" or "values". Default: "percentiles".</p></td>
    </tr>
    <tr>
      <th>point_size</th>
      <td><p>Numeric, size of points in scatterplot. Default: 1. (A smaller
value around 0.5 is better for plots which will be viewed at small scale.)</p></td>
    </tr>
    <tr>
      <th>label_repel</th>
      <td><p>Logical, if <code>label</code> is TRUE, whether to plot cluster
labels repelled from the center, on a slightly transparent white background and
with an arrow pointing to the cluster center. If FALSE, simply plot the
cluster label at the cluster center. Default: TRUE.</p></td>
    </tr>
    <tr>
      <th>label_size</th>
      <td><p>Numeric, controls the size of text labels. Default: 4.</p></td>
    </tr>
    <tr>
      <th>extra</th>
      <td><p>Logical, plot a detailed three-panel plot, where the first
is a proportional bar plot of cells in each cluster in each percentile
group, the second is a ridge plot showing density in each cluster of the mean
expression of the markers (coloured by median percentile group within the
cluster), and the third is the labelled tSNE plot coloured
by percentiles. Requires <code>label = TRUE</code>. Default: FALSE.
Default: FALSE.</p></td>
    </tr>
    <tr>
      <th>verbose</th>
      <td><p>Logical, whether to print status updates. Default: FALSE.</p></td>
    </tr>
    <tr>
      <th>hide_ticks</th>
      <td><p>Logical, whether to hide axis ticks. Default: FALSE</p></td>
    </tr>
    <tr>
      <th>label_short</th>
      <td><p>(Optional/Experimental!!) Logical, if TRUE, assumes clusters
(at seurat@ident) consist of a prefix and a suffix separated by a non-alpha
numeric character (<code>"[^[:alnum:]]+"</code>), and tries to separate these names
and only plot the prefix, for shorter labels and a cleaner plot. Default: FALSE.</p></td>
    </tr>
    </table>
    
    <h2 class="hasAnchor" id="value"><a class="anchor" href="#value"></a>Value</h2>

    <p>A ggplot object</p>
    

    <h2 class="hasAnchor" id="examples"><a class="anchor" href="#examples"></a>Examples</h2>
    <pre class="examples"><div class='input'><span class='fu'>tsneByPercentileMarkerExpression</span>(<span class='no'>pbmc</span>, <span class='st'>"IL32"</span>)</div><div class='img'><img src='tsneByPercentileMarkerExpression-1.png' alt='' width='700' height='433' /></div><div class='input'><span class='fu'>dashboard</span>(<span class='no'>pbmc</span>, <span class='st'>"IL32"</span>, <span class='kw'>title</span> <span class='kw'>=</span> <span class='st'>"Test dashboard"</span>)</div><div class='output co'>#&gt; <span class='warning'>Warning: Column `Cell` joining character vector and factor, coercing into character vector</span></div><div class='output co'>#&gt; <span class='message'>Picking joint bandwidth of 1.08</span></div><div class='img'><img src='tsneByPercentileMarkerExpression-2.png' alt='' width='700' height='433' /></div><div class='input'><span class='fu'><a href='tsneByMeanMarkerExpression.html'>feature</a></span>(<span class='no'>pbmc</span>, <span class='st'>"IL32"</span>)</div><div class='img'><img src='tsneByPercentileMarkerExpression-3.png' alt='' width='700' height='433' /></div></pre>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
    <h2>Contents</h2>
    <ul class="nav nav-pills nav-stacked">
      <li><a href="#arguments">Arguments</a></li>
      
      <li><a href="#value">Value</a></li>
      
      <li><a href="#examples">Examples</a></li>
    </ul>

    <h2>Author</h2>
    <p>Selin Jessa</p>
  </div>
</div>

      <footer>
      <div class="copyright">
  <p>Developed by Selin Jessa, Alexis Blanchet-Cohen, Yixing Hu, Claudia L. Kleinman.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.3.0.</p>
</div>
      </footer>
   </div>

  

  </body>
</html>

